cura/LayerPolygon.py

```diff
@@ -186,6 +187,11 @@ def extruder(self):
     def types(self):
         return self._types
 
+    @property
+    def lineLengths(self):
+        return [math.sqrt(sum((b - a) ** 2 for a, b in zip(self._data[i], self._data[i + 1])))
```

Comment:
You might want to consider doing this with Numpy. That is probably faster then using `zip` and list comprehensions

Reply:
457cecba5943c09b1b3b5b2a573a18c0915e0bc0